obspy/taup/tau.py

```diff
@@ -346,17 +346,16 @@ def plot_rays(self, phase_list=None, plot_type="spherical",
 
             intp = matplotlib.cbook.simple_linear_interpolation
             radius = self.model.radius_of_planet
+            phase_names_encountered = {ray.name for ray in arrivals}
```

Comment:
I wonder if we should set `colors` from `phase_names` first, then add remaining encountered ones. This way you can call `plot_rays` multiple times on the same `Axes` and get consistent colours for stuff you explicitly asked for.

Reply:
Done